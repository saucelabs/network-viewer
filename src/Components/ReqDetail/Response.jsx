import React from 'react';
import PropTypes from 'prop-types';

import Styles from './ReqDetail.styles.scss';
import CopyAllButton from './CopyAllButton';

const NoResponseText = () => (
  <h4 className={Styles['no-payload']}>This request has no response data available.</h4>
);

const Response = ({ data }) => {
  const content = data && data.body ? data.body : null;

  if (!content) {
    return <NoResponseText />;
  }

  return (
    <div className={Styles['section-detail']}>
      <div className={Styles['payload-content']}>
        <div className={Styles['copy-button']}>
          <CopyAllButton text={content} />
        </div>
        <span className={Styles['payload-body']}>
          {content}
        </span>
      </div>
    </div>
  );
};

Response.propTypes = {
  data: PropTypes.object,
};

Response.defaultProps = {
  data: null,
};

export default Response;
