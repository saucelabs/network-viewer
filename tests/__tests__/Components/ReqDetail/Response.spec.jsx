import React from 'react';
import { shallow } from 'enzyme';

import Response from './../../../../src/Components/ReqDetail/Response';

describe('Response', () => {
  it('renders without crashing for no content', () => {
    shallow(<Response data={{}} />);
  });

  it('renders without crashing for content', () => {
    shallow(<Response data={{ body: '{"foo": "bar"}' }} />);
  });
});
