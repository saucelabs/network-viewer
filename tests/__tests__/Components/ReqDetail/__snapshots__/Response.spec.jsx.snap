// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Response renders without crashing for content 1`] = `
<div>
  <div>
    <div>
      <CopyAllButton
        text="{\\"foo\\": \\"bar\\"}"
      />
    </div>
    <span>
      {"foo": "bar"}
    </span>
  </div>
</div>
`;

exports[`Response renders without crashing for no content 1`] = `<NoResponseText />`;
